src/server/admin/server/api_server.go

```diff
@@ -122,5 +130,24 @@ func (a *apiServer) InspectCluster(ctx context.Context, request *admin.InspectCl
 			response.Warnings = append(response.Warnings, fmt.Sprintf(fmtInspectProjectError, request.GetCurrentProject(), err))
 		}
 	}
+
+	if a.db != nil {
```

Comment:
Would `if a.db == nil { return response, nil }` be appropriate?

Reply:
Something I noticed is that we had an early return above that was causing these last two checks to never run.  The function should generally try to do as much as it can; so the if statements modify the response and at the very end, we just return what we have.  `a.db` shouldn't ever be nil, but I kept the provision for unit tests people might want to write in the future.